pytest_pdb.py

```diff
@@ -3,31 +3,80 @@
 import pytest
 
 
-def find_test(currentframe):
-    frame = currentframe
-    prev = frame
-    while frame:
-        for value in frame.f_locals.values():
-            if isinstance(value, pytest.Item):
-                return (value, prev)
-        prev = frame
-        frame = frame.f_back
+def find_test(stack):
+    c = 0
+    for st in stack:
+        frame = st[0]
+        c += 1
+        if frame.f_code.co_name == 'pytest_pyfunc_call':
```

Comment:
Why limit the frame here by a hardcoded name? I can't remember exactly right now, but my way of checking for any local that matches isinstance(value, pytest.Item) worked well. Do you have a specific case where it fails?

Reply:
> isinstance(value, pytest.Item)

There are several places where this is in locals().
It works better when going from the top always, but I found it not reliable (but that was when looking around), and before I've discovered that the whole stack is available.

I think it could be changed back to looking for the top-most pytest.Item maybe indeed.

Reply:
I've pushed some changes to also handle fixture setup - so a bit of name matching is necessary after all?!